### PR TITLE
Update Manage Users screen to match new designs

### LINKED ISSUE #2962: v1 Update UI and copy to reflect design
We've added a lot of functionality to the manage users page. We want to amend the UI to make it easier for admins to take the actions they need.
For v1, we will not have MFA or email/ name editing so should not include those buttons. Additionally, when a user's account is suspended, editing information, resetting MFA should be deactivated. 

See the design here: https://www.figma.com/file/NPSPM5SBzQ7Ra3mmr9M9cB/SimpleReport-Public-Site?node-id=3972%3A13760

Questions about specifics can be directed to @ashton-skylight

### PR DESCRIPTION
**Pull request by nickclyde**
## Related Issue or Background Info

- Resolves #2962

## Changes Proposed

- Change Manage Users screen UI to match designs [here](https://www.figma.com/file/NPSPM5SBzQ7Ra3mmr9M9cB/SimpleReport-Public-Site?node-id=5615%3A20890)
- Added functionality to reset a user's MFA factors

## Additional Information

- Team decided to not include the users' roles in the sidebar that lists all the users due to expensive Okta calls needed

## Screenshots / Demos

image
image
image
image
image
image
image
image
image
image

### UI
- [x] Any changes to the UI/UX are approved by design 
- [x] Any new or updated content (e.g. error messages) are approved by design 

### Testing
- [x] Includes a summary of what a code reviewer should verify

### Changes are Backwards Compatible
- [NA] Database changes are submitted as a separate PR
  - [ ] Any new tables that do not contain PII are accompanied by a GRANT SELECT to the no-PHI user
  - [ ] Any changes to tables that have custom no-PHI views are accompanied by changes to those views
        (including re-granting permission to the no-PHI user if need be)
  - [ ] Liquibase rollback has been tested locally using `./gradlew liquibaseRollbackSQL` or `liquibaseRollback`
  - [ ] Each new changeset has a corresponding [tag](https://docs.liquibase.com/change-types/community/tag-database.html)
- [x] GraphQL schema changes are backward compatible with older version of the front-end

### Security
- [NA] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [NA] Any dependencies introduced have been vetted and discussed

## Cloud
- [NA] DevOps team has been notified if PR requires ops support
- [NA] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

File: frontend/src/app/Settings/Users/ManageUsers.test.tsx [hunk 1]
```diff
@@ -224,6 +223,7 @@ let updateUserEmail: (obj: any) => Promise<any>;
 let getUsers: () => Promise<any>;
 let reactivateUser: (obj: any) => Promise<any>;
 let resetUserPassword: (obj: any) => Promise<any>;
+let resetUserMfa: (obj: any) => Promise<any>;
 let resendUserActivationEmail: (obj: any) => Promise<any>;
 
 let inputValue = (value: string) => ({ target: { value } });
```
Comment (emmastephenson) [inline frontend/src/app/Settings/Users/ManageUsers.test.tsx:229]: GH is warning that this `inputValue` seems unused

File: frontend/src/app/Settings/Users/UserFacilitiesSettingsForm.tsx [hunk 2]
```diff
@@ -127,127 +121,72 @@ const UserFacilitiesSettingsForm: React.FC<Props> = ({
...
+    <>
+      <h3 className="testing-facility-access-subheader margin-bottom-0">
+        Testing facility access{" "}
         {showRequired && <span className="text-secondary-vivid">*</span>}
       </h3>
-      <p className="text-base">{facilityAccessDescription}</p>
+      <p className="testing-facility-access-subtext">
+        {facilityAccessDescription}
+      </p>
       <Checkboxes
```
Comment (emmastephenson) [inline frontend/src/app/Settings/Users/UserFacilitiesSettingsForm.tsx:152]: Would it be possible to check this formatting against a huge list of facilities, maybe 20-30+? We have some orgs with a lot of facilities, and want to make sure they can still see/access all the facilities with the new format: https://github.com/CDCgov/prime-simplereport/issues/3089

File: frontend/src/app/Settings/Users/UsersSideNav.tsx [hunk 3]
```diff
@@ -23,9 +23,39 @@ const UsersSideNav: React.FC<Props> = ({
 }) => {
   return (
     <div className="display-block users-sidenav">
-      <h3>Users</h3>
+      <h3 className="users-header">Users</h3>
       <ul className="usa-sidenav">
         {users.map((user: LimitedUser) => {
+          let statusText;
+          switch (user.status) {
```
Comment (emmastephenson) [inline frontend/src/app/Settings/Users/UsersSideNav.tsx:30]: I remember some discussion of sorting this list by user status, @vz3 did that end up in the final designs?

**Review by emmastephenson — COMMENTED** [review]
Overall looks great!! Just a couple of small comments. Excited to see this in front of users!

The only real blocker for me is testing on an org with many facilities; everything else looks great.

**Review by jdorothy — APPROVED** [review]
This looks great! I don't really have anything to add to what @emmastephenson said other than a question: can you have an admin user without `MANAGE_USERS` permission? In the screenshots, Bob Admin can edit usernames/etc. but Kim Admin can't, despite them both being admins, unless I'm misreading this.

File: frontend/src/app/Settings/Users/UserFacilitiesSettingsForm.tsx (hunk 2 above)
Comment (emmastephenson) [reply frontend/src/app/Settings/Users/UserFacilitiesSettingsForm.tsx:152]: Thank you!! Looks great. Love the names too 😄

**Review by emmastephenson — DISMISSED** [review]
LGTM, thank you so much!

We do need to flag this to the support team before it merges in, but I can ping them directly.

**Comment by nickclyde** [comment]
The reason that's disabled for Kim Admin is because I was logged in as Kim Admin, and I have it set so that you can't edit your own settings. I wasn't sure if that was right, though - should we allow admin users to edit their own name/email?

**Comment by jdorothy** [comment]
> The reason that's disabled for Kim Admin is because I was logged in as Kim Admin, and I have it set so that you can't edit your own settings. I wasn't sure if that was right, though - should we allow admin users to edit their own name/email?

I would probably expect to be able to do that as an admin, but I know that's a little complicated for us to support. @vz3 @Rebecca-LM @emmastephenson thoughts?

**Comment by emmastephenson** [comment]
> > The reason that's disabled for Kim Admin is because I was logged in as Kim Admin, and I have it set so that you can't edit your own settings. I wasn't sure if that was right, though - should we allow admin users to edit their own name/email?
> 
> I would probably expect to be able to do that as an admin, but I know that's a little complicated for us to support. @vz3 @Rebecca-LM @emmastephenson thoughts?

That's a good question. I think I would expect that admins could do most anything to themselves except deletion, and that only because I don't know how we'd handle kicking them out of the app post-deletion. But I don't see why they shouldn't be able to take all the other actions, at least from a UX standpoint. Maybe the one catch is permissions; we don't want them to be able to make themselves a standard user and remove _all_ admins from the app by mistake.

**Comment by nickclyde** [comment]
Yeah, I'm good with enabling everything except role changes and deletion, I'll go ahead and make that change!

**Comment by nickclyde** [comment]
Alright, made it possible for admins to edit own info, and updated screenshot

**Review by jdorothy — APPROVED** [review]
LGTM - thank you!

File: frontend/src/app/Settings/Users/UserFacilitiesSettingsForm.tsx (hunk 2 above)
Comment (ashton-skylight) [reply frontend/src/app/Settings/Users/UserFacilitiesSettingsForm.tsx:152]: This looks fantastic, Nick!! +1 to the names. If the testing facility access check boxes are overwhelming, we can pivot the design where needed.

File: frontend/src/app/Settings/Users/UserDetail.tsx [hunk 4]
```diff
@@ -101,13 +101,10 @@ const UserDetail: React.FC<Props> = ({
...
-    return (
-      user.status !== "SUSPENDED" &&
-      user.status !== "PROVISIONED" &&
-      user?.id !== loggedInUser.id
-    );
-  };
+  const isUserActive = () =>
+    user.status !== "SUSPENDED" && user.status !== "PROVISIONED";
+
+  const isUserSelf = () => user.id === loggedInUser.id;
```
Comment (emmastephenson) [inline frontend/src/app/Settings/Users/UserDetail.tsx:107]: Nice, thanks for breaking this out to a helper method!

**Review by emmastephenson — APPROVED** [review]
LGTM! Support team has been notified so this should be good to merge whenever 👍

**Pull request merged**